cli/tools/task.rs

```diff
@@ -68,15 +74,55 @@ pub async fn execute_script(
   let node_resolver = factory.node_resolver().await?;
   let env_vars = task_runner::real_env_vars();
 
-  let task_runner = TaskRunner {
+  let no_of_concurrent_tasks = if let Ok(value) = std::env::var("DENO_JOBS") {
+    value.parse::<NonZeroUsize>().ok()
+  } else {
+    std::thread::available_parallelism().ok()
+  }
+  .unwrap_or_else(|| NonZeroUsize::new(2).unwrap());
+  let completed = Arc::new(Mutex::new(HashSet::with_capacity(8)));
+  let running = Arc::new(Mutex::new(HashSet::with_capacity(
+    no_of_concurrent_tasks.into(),
+  )));
+
+  let mut task_runner = TaskRunner {
     tasks_config,
     task_flags: &task_flags,
     npm_resolver: npm_resolver.as_ref(),
     node_resolver: node_resolver.as_ref(),
     env_vars,
     cli_options,
+    concurrency: no_of_concurrent_tasks.into(),
+    completed,
+    running,
+    task_names: vec![],
   };
-  task_runner.run_task(task_name).await
+
+  match task_runner.sort_tasks_topo(task_name) {
+    Ok(sorted) => task_runner.run_tasks_in_parallel(sorted).await,
+    Err(err) => match err {
+      TaskError::NotFound(name) => {
+        if task_flags.is_run {
+          return Err(anyhow!("Task not found: {}", name));
+        }
+
+        log::error!("Task not found: {}", name);
+        if log::log_enabled!(log::Level::Error) {
+          print_available_tasks(
+            &mut std::io::stderr(),
+            &task_runner.cli_options.start_dir,
+            &task_runner.tasks_config,
+          )?;
+        }
+        Ok(1)
+      }
+      TaskError::TaskDepCycle(name) => {
+        // TODO(bartlomieju): this should be improved to show the cycle
+        log::error!("Task cycle detected: {}", name);
+        Ok(1)
+      }
```

Comment:
Would be great to show the actual cycle to make it easier to debug. I think we could only compute the cycle if we hit this condition